pkg/rbd/nodeserver.go

```diff
@@ -310,25 +310,38 @@ func getLegacyVolumeName(mountPath string) (string, error) {
 }
 
 func (ns *NodeServer) mountVolumeToStagePath(ctx context.Context, req *csi.NodeStageVolumeRequest, stagingPath, devicePath string) error {
-	// Publish Path
 	fsType := req.GetVolumeCapability().GetMount().GetFsType()
 	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: mount.NewOsExec()}
+
+	// rbd images are thin-provisioned and return zeros for unwritten areas.  A freshly created
+	// image will not benefit from discard and we also want to avoid as much unnecessary zeroing
+	// as possible.  Open-code mkfs here because FormatAndMount() doesn't accept custom mkfs
+	// options.
+	//
+	// Note that "freshly" is very important here.  While discard is more of a nice to have,
+	// lazy_journal_init=1 is plain unsafe if the image has been written to before and hasn't
+	// been zeroed afterwards (unlike the name suggests, it leaves the journal completely
+	// uninitialized and carries a risk until the journal is overwritten and wraps around for
+	// the first time).
 	existingFormat, err := diskMounter.GetDiskFormat(devicePath)
 	if err != nil {
 		klog.Errorf(util.Log(ctx, "failed to get disk format for path %s, error: %v"), devicePath, err)
 		return err
 	}
-	if existingFormat == "" && (fsType == "ext4" || fsType == "xfs") {
+	// TODO: update this when adding support for static (pre-provisioned) PVs
+	if existingFormat == "" /* && !staticVol */ {
```

Comment:
Why did we remove the `fsType` check here? The discussion in the prior PR (https://github.com/ceph/ceph-csi/pull/629) was to retain the same, hence wondering why.

We do send empty `options` to `FormatAndMount`,  which in turn uses mkfs in case the format comes back as an empty string for all but `ext4` and `ext3`, so functionally we are doing the same operation.

Reply:
For clarity.  It is effectively replaced with `len(args) > 0` check below.  We still intercept only ext4 and xfs, and call `FormatAndMount` with no prior action for everything else.

Reply:
:man_facepalming: thanks!